src/benchmark_register.cc

```diff
@@ -461,16 +463,77 @@ int Benchmark::ArgsCnt() const {
   return static_cast<int>(args_.front().size());
 }
 
+void Benchmark::prepare() {}
+
 //=============================================================================//
 //                            FunctionBenchmark
 //=============================================================================//
 
+// The class used to hold all Benchmarks created from static function.
+// (ie those created using the BENCHMARK(...) macros.
+class FunctionBenchmark : public Benchmark {
+ public:
+  FunctionBenchmark(const char* name, Function* func)
+      : Benchmark(name), func_(func) {}
+
+  virtual void Run(State& st);
+
+ private:
+  Function* func_;
+};
+
 void FunctionBenchmark::Run(State& st) { func_(st); }
 
+//=============================================================================//
+//                            FixtureBenchmark
+//=============================================================================//
+//
+// The class used to hold all Benchmarks created from fixture.
+// (ie those created using the BENCHMARK_F(...) macros.
+class FixtureBenchmark : public Benchmark {
+ public:
+  FixtureBenchmark(const char* name, FixtureCreator* creator)
+      : Benchmark(name), creator_(creator) {}
+
+  void Run(State& st) override;
```

Comment:
You might as well define the functions inline.

Reply:
Inline virtual function?  The compiler has to be able to determine the pointee's type at compile time to truly inline the function, as the benchmarks are run()ed through a std::vector<std::unique_ptr<Benchmark>>, I don't think the compiler much of the function call?

Reply:
What I meant by "define inline" here is that you could put the function body *inside* the class. I don't see much value in re-declaring it only a few lines later.